components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURLBuilder.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.core;
+
+import org.wso2.carbon.identity.core.internal.DefaultServiceURLBuilder;
+import org.wso2.carbon.identity.core.internal.IdentityCoreServiceComponent;
+import org.wso2.carbon.identity.core.model.ServiceURL;
+
+/**
+ * URL Builder service interface.
+ */
+public interface ServiceURLBuilder {
+
+    static ServiceURLBuilder create() {
+
+        if (IdentityCoreServiceComponent.getServiceURLBuilderFactory() != null) {
+            return IdentityCoreServiceComponent.getServiceURLBuilderFactory().getServiceURLBuilder();
+        } else {
+            return new DefaultServiceURLBuilder();
+        }
+
```

Comment:
remove unnecessary line.
Also, hope we checked that factory gets resolved from OSGi service level

Reply:
Yes, created a dummy method and a test in IdentityUtil and checked this.